utils/python/CIME/hist_utils.py

```diff
@@ -123,9 +124,33 @@ def _hists_match(model, hists1, hists2, suffix1="", suffix2=""):
     two_not_one = sorted([hists2[normalized2.index(item)] for item in set_of_2_not_1])
 
     both = set(normalized1) & set(normalized2)
+
     match_ups = sorted([ (hists1[normalized1.index(item)], hists2[normalized2.index(item)]) for item in both])
-    expect(len(match_ups) + len(set_of_1_not_2) == len(hists1), "Programming error1")
-    expect(len(match_ups) + len(set_of_2_not_1) == len(hists2), "Programming error2")
+    if multiinst:
+        new_two_not_one = []
+        new_one_not_two = []
+        for normalized_name in two_not_one:
+            m = re.search(r"(.*%s.*)_[0-9]{4}(.h.*)"%model, normalized_name)
+            if m is not None:
+                multi_normalized = m.group(1)+m.group(2)
+                if multi_normalized in one_not_two:
+                    match_ups.append((multi_normalized, normalized_name) )
+                else:
+                    new_two_not_one.append(normalized_name)
+        two_not_one = new_two_not_one
+        for normalized_name in one_not_two:
+            m = re.search(r"(.*%s.*)_[0-9]{4}(.h.*)"%model, normalized_name)
+            if m is not None:
+                multi_normalized = m.group(1)+m.group(2)
+                if multi_normalized in two_not_one:
+                    match_ups.append((multi_normalized, normalized_name) )
+                else:
+                    new_one_not_two.append(normalized_name)
+        one_not_two = new_one_not_two
+
+    else:
+        expect(len(match_ups) + len(set_of_1_not_2) == len(hists1), "Programming error1")
```

Comment:
Is it possible to have similar 'expect' statements for the multi-instance case? Checking these invariants seems important given the complexity of what's going on here, especially in the multi-instance case.